server/src/main/java/org/opensearch/common/util/FeatureFlags.java

```diff
@@ -60,6 +60,11 @@ public class FeatureFlags {
      */
     public static final String DATETIME_FORMATTER_CACHING = "opensearch.experimental.optimization.datetime_formatter_caching.enabled";
 
+    /**
+     * Gates the query categorization by type feature.
+     */
```

Comment:
Since feature flags are static in nature, these flags shouldn't be a feature flag.  These should ideally be a dynamic flag and should come from the cluster settings.

Reply:
done